### PR TITLE
Updated Maxbin2 for uncompressing read files if .gz files are given

### PR DESCRIPTION
**Pull request by Ralwoss**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/maxbin2/maxbin2.xml [hunk 1]
```diff
@@ -25,13 +25,31 @@
...
-echo '$assembly.inputs.reads' >> reads_list &&
+        ## uncompress .gz reads files if necessary
+        #set $e = $assembly.inputs.reads
+        #set $identifier = re.sub('[^\s\w\-\\.]', '_', str($e.element_identifier))
+        #if $e.ext.endswith("fastq.gz") or $e.ext.endswith("fastqsanger.gz") or $e.ext.endswith("fasta.gz")
+            #set $newid = $identifier.removesuffix('.gz')
+gunzip -c '$e' > '$newid' &&
+echo '$newid' >> reads_list &&
+        #else 
+ln -s '$e' '${identifier}' &&
```
Comment (bgruening) [inline tools/maxbin2/maxbin2.xml:39]: why do we have here a link and below not?

**Comment by bgruening** [comment]
Can you please also bump the version. Thanks @Ralwoss

File: tools/maxbin2/maxbin2.xml (hunk 1 above)
Comment (Ralwoss) [reply tools/maxbin2/maxbin2.xml:39]: With multiple read files there were errors when using a link if the same file is used multiple times (like in the test cases)

File: tools/maxbin2/maxbin2.xml [hunk 2]
```diff
@@ -94,7 +112,7 @@ run_MaxBin.pl
                 <conditional name="inputs">
                     <expand macro="input_type"/>
                     <when value="reads">
-                        <param argument="-reads" type="data" format="fasta,fastq" label="Reads file"/>
+                        <param argument="-reads" type="data" format="fasta,fastq,fastq.gz" label="Reads file"/>
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:115]: Could you add also `fasta.gz` here? Thanks

File: tools/maxbin2/maxbin2.xml [hunk 3]
```diff
@@ -25,13 +25,31 @@
...
     <command detect_errors="exit_code"><![CDATA[
 ## generate read or abundance files
+#import re
 #if $assembly.inputs.type == 'reads'
     #if $assembly.type == 'individual'
-echo '$assembly.inputs.reads' >> reads_list &&
+        ## uncompress .gz reads files if necessary
+        #set $e = $assembly.inputs.reads
+        #set $identifier = re.sub('[^\s\w\-\\.]', '_', str($e.element_identifier))
+        #if $e.ext.endswith("fastq.gz") or $e.ext.endswith("fastqsanger.gz") or $e.ext.endswith("fasta.gz")
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:34]:
```suggestion
        #if $e.ext.endswith(".gz")
```

File: tools/maxbin2/maxbin2.xml [hunk 4]
```diff
@@ -25,13 +25,31 @@
...
+echo '$newid' >> reads_list &&
+        #else 
+ln -s '$e' '${identifier}' &&
+echo '$identifier' >> reads_list &&
+        #end if
     #else
         #for $r in $assembly.inputs.reads
             #if $r
+                #set $identifier = re.sub('[^\s\w\-\\.]', '_', str($r.element_identifier))
+                #if $r.ext.endswith("fastq.gz") or $r.ext.endswith("fastqsanger.gz") or $r.ext.endswith("fasta.gz")
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:46]:
```suggestion
                #if $r.ext.endswith(".gz")
```

File: tools/maxbin2/maxbin2.xml [hunk 5]
```diff
@@ -25,13 +25,31 @@
...
+                    #set $newid = $identifier.removesuffix('.gz')
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:47]: I do not think you might have to remove `.gz` there. `.gz` will appear if user puts `.gz` in the history element name

File: tools/maxbin2/maxbin2.xml (hunk 1 above)
Comment (bebatut) [reply tools/maxbin2/maxbin2.xml:39]: You could use the index in the for loop to add in the new name. Something like here: https://github.com/galaxyproject/tools-iuc/blob/main/tools/coverm/macros.xml#L227

**Review by bebatut — COMMENTED** [review]
Thanks @Ralwoss. I put some comments. Let me know if you need more explanation

File: tools/maxbin2/maxbin2.xml [hunk 6]
```diff
@@ -25,13 +25,32 @@
...
-echo '$assembly.inputs.reads' >> reads_list &&
+        ## uncompress .gz reads files if necessary
+        #set $e = $assembly.inputs.reads
+        #set $identifier = re.sub('[^\s\w\-\\.]', '_', str($e.element_identifier))
+        #if $e.ext.endswith(".gz")
+            #set $newid = $identifier
+gunzip -c '$e' > '$newid' &&
+echo '$newid' >> reads_list &&
+        #else 
+ln -s '$e' '${identifier}' &&
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:39]:
```suggestion
ln -s '$e' '$identifier' &&
```

File: tools/maxbin2/maxbin2.xml [hunk 7]
```diff
@@ -25,13 +25,32 @@
...
+echo '$identifier' >> reads_list &&
+        #end if
     #else
-        #for $r in $assembly.inputs.reads
+        #for $i, $r in enumerate($assembly.inputs.reads)
             #if $r
-echo '$r' >> reads_list &&
+                #set $identifier = re.sub('[^\s\w\-\\.]', '_', str($r.element_identifier))
+                #set $newid = 'reads' + str($i) + '_' + $identifier
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:46]:
```suggestion
                #set $newid = $identifier +  '_'  + str($i)
```

File: tools/maxbin2/maxbin2.xml [hunk 8]
```diff
@@ -1,8 +1,5 @@
-<tool id="maxbin2" name="MaxBin2" version="@MAXBIN_VERSION@+galaxy3">
+<tool id="maxbin2" name="MaxBin2" version="@MAXBIN_VERSION@+galaxy4">
     <description>clusters metagenomic contigs into bins</description>
-    <xrefs>
-        <xref type="bio.tools">masigpro</xref>
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:4]: Please leave that but replace masigpro with maxbin

File: tools/maxbin2/maxbin2.xml [hunk 9]
```diff
@@ -19,19 +16,40 @@
             <param argument="--reassembly" type="boolean" truevalue="-reassembly" falsevalue="" checked="false" label="Reassembly" help="Reassembly option is still highly experimental. To use this function, you need to feed MaxBin interleaved paired-end fastq or fasta file." />
         </xml>
     </macros>
+    <xrefs>
+        <xref type="bio.tools">masigpro</xref>
```
Comment (bebatut) [inline tools/maxbin2/maxbin2.xml:20]:
```suggestion
        <xref type="bio.tools">maxbin</xref>
```

File: tools/maxbin2/maxbin2.xml (hunk 2 above)
Comment (bebatut) [reply tools/maxbin2/maxbin2.xml:115]:
```suggestion
                        <param argument="-reads" type="data" format="fasta,fastq,fastq.gz,fasta.gz" label="Reads file"/>
```

**Review by bebatut — APPROVED** [review]
Thanks @Ralwoss!

**Pull request merged**